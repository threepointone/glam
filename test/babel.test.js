/* eslint-disable no-useless-escape */
/* eslint-env jest */
const babel = require('babel-core')
const glamPlugin = require('../src/babel')
const { flush } = require('../src')

describe('babel plugin', () => {
  beforeEach(flush)
  test('normal', () => {
    const basic = `
      css\`
       margin: 12px 48px;
       color: #ffffff;
       height: \$\{props => props.height * props.scale\};
       display: flex;
       flex: 1 0 auto;
       color: blue;
       width: \$\{widthVar\};
    \``
    const { code } = babel.transform(basic, {
      plugins: [[glamPlugin, { sync: true }]]
    })
    expect(code).toMatchSnapshot()
  })

  test('inline simple', () => {
    const basic = `
      css\`
       margin: 12px;
       color: \$\{colorVar};
       height: \$\{heightVar};
      \`
    `
    const { code } = babel.transform(basic, {
      plugins: [[glamPlugin, { sync: true, inline: true }]]
    })
    expect(code).toMatchSnapshot()
  })

  test('inline kitchen sink', () => {
    const basic = `
      const cls = css\`font-size: 58pt;margin: $\{margin};\`
      const frag = fragment\`padding: 8px;\`
      const fragB = fragment\`height: $\{heightVar};@apply \$\{frag};\`
      const cls2 = css\`
       @apply $\{frag};
       @apply $\{fragB}; 
       margin: 12px 48px;
       color: #ffffff;
      \`
    `
    const { code } = babel.transform(basic, {
      plugins: [[glamPlugin, { sync: true, inline: true }]]
    })
    expect(code).toMatchSnapshot()
  })
})
