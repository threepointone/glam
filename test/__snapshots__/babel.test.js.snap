// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel plugin inline kitchen sink 1`] = `
"
const cls = css(\\"css-h5t3cu\\", [margin], function inlineCss(x0) {
  return [\`.css-h5t3cu { font-size: 58pt;margin: \${x0}; }\`];
});
const frag = fragment('frag-zzm2ov', [], () => [\`.frag-zzm2ov { padding: 8px; }\`]);
const fragB = fragment('frag-16v04hq', [heightVar, frag], (x0, x1) => [\`.frag-16v04hq { height: \${x0};\${x1}; }\`]);
const cls2 = css(\\"css-194c4h8\\", [frag, fragB], function inlineCss(x0, x1) {
  return [\`.css-194c4h8 { \${x0};
       \${x1}; 
       margin: 12px 48px;
       color: #ffffff; }\`];
});"
`;

exports[`babel plugin inline simple 1`] = `
"
css(\\"css-r1aqtk\\", [colorVar, heightVar], function inlineCss(x0, x1) {
  return [\`.css-r1aqtk { margin: 12px;
       color: \${x0};
       height: \${x1}; }\`];
});"
`;

exports[`babel plugin normal 1`] = `
"
css(\\"css-186afxg\\", [props => props.height * props.scale, widthVar]);"
`;
